solutions/silver/cf-1777C.mdx

```diff
@@ -0,0 +1,126 @@
+---
+id: cf-1777C
+source: CF
+title: Quiz Master
+author: Rameez Parwez
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/111729)
+
+## Explanation
+
+Let's use a sliding window technique on sorted smartness values to track how many topics are being covered as we add and remove a student.
+As we expand the team by moving the right pointer, we check the factors of each student's smartness to determine which topics they help cover.
+If all topics are covered, we calculate the difference between the highest and lowest smartness values in the current team.
+Then, we attempt to reduce the size of the team from the left to see if the range can be further minimized while still covering all topics.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(n \log(n) + n \cdot \sqrt{M})$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
```

Comment:
if you're gonna not use `namespace std` might as well also not use `bits/stdc++`